packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -682,19 +683,45 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
     return linkMarks;
   }
 
+  isValidHref(url: string): boolean {
+    if (!this._autoLinkRegexNonGlobal?.test(url)) {
+      return false;
+    }
+
+    const href = this.buildHref(url);
+
+    if (!this.isValidTLD(href)) {
+      return false;
+    }
+
+    return true;
```

Comment:
Not sure you need to include this change with this particular PR. It was an attempt to remove previously invalid links but it doesn't really work as we'd hoped so maybe we should just focus on the `autoLinkAfter` change in this PR?